istio/assets/configuration/spec.yaml

```diff
@@ -38,56 +37,7 @@ files:
             display_default: null
             example: http://istio-proxy.istio-system:15090/stats/prometheus
             type: string
-        - name: mixer_endpoint
-          description: |
-            Define the mixer endpoint in order to collect all Prometheus metrics on the Mixer process as well
-            as gRPC metrics related to API calls and metrics on adapter dispatch.
-            If using Istio < v1.1, replace port 15014 with port 9093. See the changes here:
-            https://istio.io/about/notes/1.1/#configuration-management
-          value:
-            display_default: null
-            example: http://istio-telemetry.istio-system:15014/metrics
-            type: string
-        - name: pilot_endpoint
-          description: |
-            Define the pilot endpoint in order to collect all Prometheus metrics on the Pilot process.
-            Only available for Istio >= v1.1.
-          value:
-            display_default: null
-            example: http://istio-pilot.istio-system:15014/metrics
-            type: string
-        - name: galley_endpoint
-          description: |
-            Define the galley endpoint in order to collect all Prometheus metrics on the Galley process.
-            Only available for Istio >= v1.1.
-          value:
-            display_default: null
-            example: http://istio-galley.istio-system:15014/metrics
-            type: string
-        - name: citadel_endpoint
-          description: |
-            Define the citadel endpoint in order to collect all Prometheus metrics on the Citadel process.
-            Only available for Istio >= v1.1.
-          value:
-            display_default: null
-            example: http://istio-citadel.istio-system:15014/metrics
-            type: string
-        - template: instances/openmetrics_legacy
+        - template: instances/openmetrics
           overrides:
-            prometheus_url.hidden: true
-            exclude_labels.enabled: true
-            exclude_labels.value.example:
-              - source_version
-              - destination_version
-              - source_canonical_revision
-              - destination_canonical_revision
-              - source_principal
-              - destination_principal
-              - source_cluster
-              - destination_cluster
-              - source_canonical_service
-              - destination_canonical_service
-              - source_workload_namespace
-              - destination_workload_namespace
-              - request_protocol
-              - connection_security_policy
+            openmetrics_endpoint.required: false
+            openmetrics_endpoint.hidden: true
```

Comment:
Can you add the following so config models can still validate options but we don't document them in conf.yaml.example? I added logic to hide entire templates.
 
https://github.com/DataDog/integrations-core/blob/master/openmetrics/assets/configuration/spec.yaml#L16-L19

We also need to add this to config models https://github.com/DataDog/integrations-core/blob/master/openmetrics/datadog_checks/openmetrics/config_models/validators.py

Please test with dd_run_check to make sure nothing breaks with either `use_openmetrics` enabled or disabled

Reply:
There's already an existing `validators.py` for `istio`. I tried using the one for `openmetrics` that you linked, and I don't think this would work for `istio`. The `istio` implementation allows `istiod_endpoint` and `istio_mesh_endpoint` to be used instead of `prometheus_url` or `openmetrics_endpoint`, since the configs would generate automatically instead. The tests simulate this behavior, so using `openmetrics`'s validator causes errors.

Reply:
Right, I didn't mean copy+paste that example but to verify we don't need to add additional validator for the `use_openmetrics` option.